db/merge_helper.cc

```diff
@@ -348,32 +525,31 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
           c_iter_stats->total_blob_bytes_read += bytes_read;
         }
 
-        s = TimedFullMerge(user_merge_operator_, ikey.user_key, &blob_value,
-                           merge_context_.GetOperands(), &merge_result, logger_,
-                           stats_, clock_,
-                           /* result_operand */ nullptr,
-                           /* update_num_ops_stats */ false, &op_failure_scope);
+        s = TimedFullMerge(user_merge_operator_, ikey.user_key, kPlainBaseValue,
+                           blob_value, merge_context_.GetOperands(), logger_,
+                           stats_, clock_, /* update_num_ops_stats */ false,
+                           &merge_result, /* result_operand */ nullptr,
+                           &merge_result_type, &op_failure_scope);
       } else if (ikey.type == kTypeWideColumnEntity) {
-        s = TimedFullMergeWithEntity(
-            user_merge_operator_, ikey.user_key, iter->value(),
-            merge_context_.GetOperands(), &merge_result, logger_, stats_,
-            clock_, /* update_num_ops_stats */ false, &op_failure_scope);
+        s = TimedFullMerge(user_merge_operator_, ikey.user_key, kWideBaseValue,
+                           iter->value(), merge_context_.GetOperands(), logger_,
+                           stats_, clock_, /* update_num_ops_stats */ false,
+                           &merge_result, /* result_operand */ nullptr,
+                           &merge_result_type, &op_failure_scope);
       } else {
-        s = TimedFullMerge(user_merge_operator_, ikey.user_key, nullptr,
-                           merge_context_.GetOperands(), &merge_result, logger_,
-                           stats_, clock_,
-                           /* result_operand */ nullptr,
-                           /* update_num_ops_stats */ false, &op_failure_scope);
+        s = TimedFullMerge(user_merge_operator_, ikey.user_key, kNoBaseValue,
+                           merge_context_.GetOperands(), logger_, stats_,
+                           clock_, /* update_num_ops_stats */ false,
+                           &merge_result, /* result_operand */ nullptr,
+                           &merge_result_type, &op_failure_scope);
       }
 
       // We store the result in keys_.back() and operands_.back()
       // if nothing went wrong (i.e.: no operand corruption on disk)
       if (s.ok()) {
         // The original key encountered
         original_key = std::move(keys_.back());
-        orig_ikey.type = ikey.type == kTypeWideColumnEntity
-                             ? kTypeWideColumnEntity
-                             : kTypeValue;
+        orig_ikey.type = merge_result_type;
```

Comment:
`assert(merge_result_type == kTypeValue || merge_result_type == kTypeWideColumnEntity);`

Wondering if we want to assert value result types before setting it here.

Reply:
Good idea! Will do